### DISCUSSION TITLE
Blackpoint compensation (BPC) when performing icc-conversion with libvips?

### DISCUSSION THREAD

**Discussion by hanssonrickard** [libvips/libvips#2164] [Q&A] [answered]
Hi.
I am trying out libvips together with python3 (pyvips) and it seems to work very well.
Only thing missing is that i cannot find any info if libvips support "blackpoint compensation" as an option when performing icc conversion.

Do i understand it correct that the icc conversion is based on lcms? 
Afaik lcms has support for Blackpoint compensation.

As of now the colors are correct if i compare with conversion done in Photoshop, but only if i compare with BPC disabled in Photoshop conversion.

**Comment by jcupitt** [comment]
Hello @hanssonrickard, 

No libvips does not expose any of lcms's blackpoint compensation controls, just "intent". Have you tried using perceptual instead? It might do what you need.

Why BPC is bad:

https://support.captureone.com/hc/en-us/articles/360002654477-Black-point-compensation

**Reply by hanssonrickard** [reply]
Thanks for the reply.
No, perceptual is not an option for our customers in the graphical industry that prints CMYK, for them it is "relative" rendering that is used for almost 100% of all conversions to CMYK (or absolute if they are printing proofs) because it is crucial that all colors that fits inside the output gamut are printed as accurate as possible.

Yeah, i do not like to use BPC and most of our large format printers does not use it either, but for some customers that have been using it for a long time and wish to get similar result, it is of interest to be able to use/enable it regardless of which software solution they have.

Anyhow, it it is not posible, then that is the way to go and i can/will just have to tell the customers that they will have to live without it.

**Reply by jcupitt** [reply]
It would be easy to add, I'll make a PR.

**Comment by jcupitt** [comment, marked as answer]
There's a PR with this feature. You can now do:

```
$ vips icc_transform cmyk2.jpg x.jpg srgb --black-point-compensation
```

Or in Python:

```
result = image.icc_transform("srgb", black_point_compensation=True)
```

Any testing etc. very welcome.

**Reply by jcupitt** [reply]
PR here https://github.com/libvips/libvips/pull/2171

**Reply by hanssonrickard** [reply]
Thanks.
Maybe a dumb question, but what do i have to do to test it? I need to download the source and compile it?
As of today i have it installed using package from Debian 9 package list.

apt-cache search libvips-tools
libvips-tools - image processing system good for very large ones (tools)

**Comment by jcupitt** [comment]
Yes, you'd need to compile from source.
